File: src/csharp/Intel/Iced/Intel/BlockEncoderInternal/CodeWriterImpl.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if ENCODER && BLOCK_ENCODER
namespace Iced.Intel.BlockEncoderInternal {
	sealed class CodeWriterImpl : CodeWriter {
		public uint BytesWritten;
		readonly CodeWriter codeWriter;

		public CodeWriterImpl(CodeWriter codeWriter) {
			if (codeWriter is null)
				ThrowHelper.ThrowArgumentNullException_codeWriter();
			this.codeWriter = codeWriter;
		}

		public override void WriteByte(byte value) {
			BytesWritten++;
			codeWriter.WriteByte(value);
		}
	}
}
#endif

File: src/java/iced-x86/src/main/java/com/github/icedland/iced/x86/enc/CodeWriterImpl.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.enc;

import com.github.icedland.iced.x86.CodeWriter;

final class CodeWriterImpl implements CodeWriter {
	public int bytesWritten;
	final CodeWriter codeWriter;

	public CodeWriterImpl(CodeWriter codeWriter) {
		if (codeWriter == null)
			throw new NullPointerException("codeWriter");
		this.codeWriter = codeWriter;
	}

	@Override
	public void writeByte(byte value) {
		bytesWritten++;
		codeWriter.writeByte(value);
	}
}
